penny_university_frontend/src/components/modal/auth/email.tsx

```diff
@@ -2,29 +2,28 @@
 
 import React, { useState } from 'react'
 import { connect } from 'react-redux'
+import { ThunkDispatch } from 'redux-thunk'
+import { AnyAction } from 'redux'
 import {
   Form, ModalHeader, ModalBody, Button,
 } from 'reactstrap'
-import propTypes from 'prop-types'
 import { Input } from '../../fields'
 import modalDispatch from '../dispatch'
 import { dispatchUserExists } from '../../../actions/user'
 
-const AuthEmailModal = ({ userExists }) => {
+type AuthEmailModalProps = {
+  userExists: (email: string) => Promise<AnyAction>
+}
+
+const AuthEmailModal = ({ userExists }: AuthEmailModalProps) => {
   const [email, setEmail] = useState('')
   return (
     <>
       <ModalHeader toggle={modalDispatch.close}>Sign up or log in</ModalHeader>
       <ModalBody>
         <Form onSubmit={(e) => {
           e.preventDefault()
-          userExists(email).then((action) => {
-            if (action.payload?.status === 400) {
-              modalDispatch.authSignup(email)
-            } else {
-              modalDispatch.authPassword(email)
-            }
-          })
+          userExists(email)
```

Comment:
This became obvious to me how to clean this up without catching the dispatch promise in the view. Handle in the middleware!